cdap-watchdog/src/main/java/io/cdap/cdap/logging/clean/LogCleaner.java

```diff
@@ -130,16 +120,16 @@ private void cleanupLogFolders() {
    *         folders are cleaned up, true otherwise.
    * @throws IOException
    */
-  private boolean cleanupEmptyFolders(@Nullable Location location, Set<String> excludeFolders) throws IOException {
+  private boolean cleanupEmptyFolders(@Nullable Location location, long checkpointTimestamp) throws IOException {
     if (location == null) {
       return true;
     }
     if (!location.isDirectory()) {
       // Parent directory has files.
       return false;
     }
-    if (excludeFolders.contains(location.getName())) {
-      // No need to traverse excluded folders
+    if (location.lastModified() > checkpointTimestamp) {
```

Comment:
What would be the last modified date of a directory Location?

Reply:
Last modified date of a directory is the date at which the directory was created or any file within that directory was modified (or created / deleted).